src/main/java/gyro/google/compute/RegionDiskResource.java

```diff
@@ -0,0 +1,268 @@
+ *         source-snapshot-encryption-key
+ *         kms-key-name: "my-kms-key-name"
+ *         end
+ *     end
+ */
+@Type("compute-region-disk")
+public class RegionDiskResource extends AbstractDiskResource {
+    private String region;
+    private List<String> replicaZones;
+
+    /**
+     * The region where the disk resides. (Required)
+     */
+    @Required
+    public String getRegion() {
+        return region;
+    }
+
+    public void setRegion(String region) {
+        this.region = region;
+    }
+
+    /**
+     * The zones where the disk should be replicated to. (Required)
+     */
+    @Required
+    public List<String> getReplicaZones() {
+        if (replicaZones == null) {
+            replicaZones = new ArrayList<>();
+        }
+        return replicaZones;
+    }
+
+    public void setReplicaZones(List<String> replicaZones) {
+        this.replicaZones = replicaZones != null
+            ? replicaZones.stream().map(zone -> ComputeUtils.toZoneUrl(getProjectId(), zone)).collect(Collectors.toList())
+            : null;
+    }
+
+    @Override
+    public void setResourcePolicies(List<String> resourcePolicies) {
+        super.setResourcePolicies(resourcePolicies != null
+            ? resourcePolicies.stream().map(p -> ComputeUtils.toResourcePolicyUrl(getProjectId(), p, getRegion())).collect(Collectors.toList())
+            : null);
+    }
+
+    @Override
+    public void setType(String type) {
+        super.setType(type != null ? ComputeUtils.toRegionDiskTypeUrl(getProjectId(), type, getRegion()) : null);
+    }
+
+    @Override
+    public void copyMore(Disk disk) {
+        setRegion(disk.getRegion().substring(disk.getRegion().lastIndexOf("/") + 1));
+        setReplicaZones(disk.getReplicaZones());
+        setType(disk.getType());
+        setResourcePolicies(disk.getResourcePolicies());
+    }
+
+    @Override
+    public void doCreate(GyroUI ui, State state, Disk disk) {
+        Compute client = createComputeClient();
+
+        disk.setRegion(getRegion());
+        disk.setReplicaZones(getReplicaZones());
+
+        try {
+            Compute.RegionDisks.Insert insert = client.regionDisks().insert(getProjectId(), getRegion(), disk);
+            Operation operation = insert.execute();
+            Operation.Error error = waitForCompletion(client, operation);
+            if (error != null) {
+                throw new GyroException(error.toPrettyString());
+            }
+
+            refresh();
+        } catch (Exception ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public List<ValidationError> validateMore() {
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (getReplicaZones().size() != 2) {
+            errors.add(new ValidationError(
+                this,
+                "replica-zones",
+                "Disk requires exactly two replica zones."));
+        }
+
+        return errors;
+    }
```

Comment:
Rather than making an abstract `validateMore()` method that all concrete class need to implement, the concrete class can call its own validate method.

```
public List<ValidationError> validate() {
        List<ValidationError> errors = super.validate();

        if (getReplicaZones().size() != 2) {
            errors.add(new ValidationError(
                this,
                "replica-zones",
                "Disk requires exactly two replica zones."));
        }

        return errors;
    }
```